server/src/main/java/org/opensearch/indices/recovery/RecoveryTarget.java

```diff
@@ -435,6 +435,27 @@ public void writeFileChunk(
         }
     }
 
+    /**
+     * fail the recovery and call listener
+     *
+     * @param e                exception that encapsulates the failure
+     * @param sendShardFailure indicates whether to notify the master of the shard failure
+     */
+    public void fail(RecoveryFailedException e, boolean sendShardFailure) {
```

Comment:
With `RecoveryFailedException` extending `ReplicationFailedException` I don't think we need to override here in RecoveryTarget?

Reply:
Yeah, makes sense. This is redundant now, so will remove it.